presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -891,6 +892,11 @@ public Optional<String> getExpandedQuery()
         return expandedQuery;
     }
 
+    public Optional<String> getQueryTemplate()
+    {
+        return Optional.of(formatSqlAsTemplate(root, Optional.empty()));
```

Comment:
Can we set this into a variable in analysis so we don't do duplicate work if this is called more than once? Maybe we can add a setter and set it from StatementAnalyzer?

Reply:
Updated to add a setter for query template in analysis, and to set from Analyzer as discussed offline.